pixi_docs/src/main.rs

```diff
@@ -110,43 +117,47 @@ fn subcommand_to_md(parents: &[String], command: &Command) -> String {
         }
     }
 
+    // Get a list of the parents and command name
+    let mut parts = parents.to_vec();
+    parts.push(command.get_name().to_string());
+
     // Positionals
     let positionals: Vec<_> = command.get_positionals().collect();
     if !positionals.is_empty() {
         writeln!(buffer, "\n## Arguments").unwrap();
-        write!(buffer, "{}", arguments(&positionals)).unwrap();
+        write!(buffer, "{}", arguments(&positionals, &parts)).unwrap();
     }
 
     // Options
     let opts: Vec<_> = command.get_opts().collect();
     if !opts.is_empty() {
-        let sorted_opts: Vec<_> = opts
-            .into_iter()
-            .filter(|o| !o.is_hide_set())
-            .sorted_by(|a, b| {
-                a.get_long()
-                    .unwrap_or_default()
-                    .cmp(&b.get_long().unwrap_or_default())
+        // All options with their header
+        let header_option_map: HashMap<String, Vec<&Arg>> = opts
+            .iter()
+            .map(|o| {
+                (
+                    o.get_help_heading().unwrap_or("Options").to_string(),
+                    o.to_owned(),
+                )
             })
-            .collect();
-
-        // Split into global and non-global options
-        let (global_opts, regular_opts): (Vec<_>, Vec<_>) =
-            sorted_opts.into_iter().partition(|o| o.is_global_set());
-
-        // Regular (non-global) options
-        if !regular_opts.is_empty() || command.get_version().is_some() {
-            writeln!(buffer, "\n## Options").unwrap();
-            if command.get_version().is_some() {
-                writeln!(buffer, "- <a id=\"option-version\" href=\"#option-version\">`--version (-V)`</a>\n: Display version information").unwrap();
+            .into_group_map();
+
+        let sorted_header_options = header_option_map.iter().sorted_by(|a, b| {
```

Comment:
For reviewer:
This is the logic to make the headers seperated, and always make the `Options` the first, and `Global Options` last.